src/main/java/spring/intro/controller/UserController.java

```diff
@@ -0,0 +1,44 @@
+package spring.intro.controller;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import spring.intro.dto.UserResponseDto;
+import spring.intro.model.User;
+import spring.intro.service.UserService;
+
+@RestController
+@RequestMapping("/user")
+public class UserController {
+    private final UserService userService;
+
+    @Autowired
+    public UserController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @GetMapping("/inject")
+    public void injectData() {
+        userService.add(new User("nikita@mail.com", "1234"));
+        userService.add(new User("vaganov@mail.com", "5678"));
+    }
+
+    @GetMapping("/{userId}")
+    public UserResponseDto get(@PathVariable Long userId) {
+        return mapToDto(userService.get(userId));
+    }
+
+    @GetMapping
+    public List<UserResponseDto> getAll() {
+        return userService.listUsers().stream().map(this::mapToDto)
+                .collect(Collectors.toList());
+    }
+
+    private UserResponseDto mapToDto(User user) {
+        return new UserResponseDto(user.getId(), user.getEmail(), user.getPassword());
```

Comment:
From the checklist "Don’t pass User in the constructor of UserResponseDto, let’s use setters."